pkg/node/requester.go

```diff
@@ -209,11 +215,21 @@ func NewRequesterNode(
 		translationProvider = translation.NewStandardTranslatorsProvider()
 	}
 
+	instanceID, err := metaStore.ReadInstanceID()
+	if err != nil {
+		return nil, err
+	}
+	installationID, err := metaStore.ReadInstallationID()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is a common pattern I am seeing in different places in the code where we should treat these as best effort instead of failing. My recommendation is to have the metastore handle the failures internally by just logging and returning an empty id. If there are cases where you need to fail, then you can introduce `MustReadInstanceID()` methods.

What would be nice is:
1. Have the metastore read the ids from disk only once the first time a getter is called, store the entries in-memory and then serve subsequent requests from memory
2. If a failure happens while reading from disk, then it should log only once, and then all future calls will just get an empty id without more logging

Reply:
I have modified this slightly such that the entire metadata struct is returned rather than reading individual fields.